tests/python/pants_test/backend/codegen/tasks/test_simple_codegen_task.py

```diff
@@ -0,0 +1,185 @@
+        return 'global'
+
+    task = self._create_subclass_task(TestGen, strategy='global')
+    self.assertEqual('global', task.codegen_strategy)
+    task = self._create_subclass_task(TestGen, strategy='isolated')
+    self.assertEqual('global', task.codegen_strategy)
+
+  def test_codegen_workdir_suffix(self):
+    target_specs = [
+      'project/src/main/foogen/foo-lib:foo-target-a',
+      'project/src/main/foogen/foo-lib:foo-target-b',
+      'project/src/main/foogen/foo-bar:foo-target-a',
+      'project/src/main/genfoo/foo-bar:foo-target-a',
+    ]
+    spec_paths = set([spec.split(':')[0] for spec in target_specs])
+    spec_names = set([spec.split(':')[1] for spec in target_specs])
+    for spec_path in spec_paths:
+      parent_dir = os.path.dirname(spec_path)
+      if not os.path.exists(parent_dir):
+        self.create_dir(parent_dir)
+      self.create_file('{0}/BUILD'.format(spec_path), '\n'.join([dedent('''
+          java_protobuf_library(name='{name}',
+            sources=[],
+            dependencies=[],
+          )
+        '''.format(name=spec_name)) for spec_name in spec_names]))
+
+    targets = set([self.target(spec) for spec in target_specs])
+
+    get_suffix = SimpleCodegenTask._codegen_workdir_suffix
+
+    for target in targets:
+      self.assertEqual('global', get_suffix(target, 'global'))
+      self.assertTrue('isolated' in get_suffix(target, 'isolated'))
+
+    global_dirs = set(get_suffix(target, 'global') for target in targets)
+    isolated_dirs = set(get_suffix(target, 'isolated') for target in targets)
+
+    self.assertEqual(1, len(global_dirs), 'There should only be one global directory suffix!')
+    self.assertEqual(len(targets), len(isolated_dirs),
+                     'There should be exactly one directory suffix per unique target!')
+
+    for target in targets:
+      for strategy in ('global', 'isolated',):
+        suffix = None
+        for i in range(5):
+          curr_suffix = get_suffix(target, strategy)
+          if i > 0:
+            self.assertEqual(suffix, curr_suffix,
+                             'Suffix should not be unstable for the same target and strategy! '
+                             '({target} with {strategy} got {suffix_a} and {suffix_b})'
+                             .format(target=target.id,
+                                     strategy=strategy,
+                                     suffix_a=suffix,
+                                     suffix_b=curr_suffix))
+          suffix = curr_suffix
+
+  def test_execute(self):
+    proto_suffixes = ['a', 'b', 'c',]
+
+    self.add_to_build_file('gen-lib', '\n'.join(dedent('''
+      java_wire_library(name='{suffix}',
+        sources=['org/pantsbuild/example/foo{suffix}.proto'],
+      )
+    ''').format(suffix=suffix) for suffix in proto_suffixes))
+
+    for suffix in proto_suffixes:
+      self.create_file('gen-lib/org/pantsbuild/example/foo{suffix}.proto'.format(suffix=suffix),
```

Comment:
This seems like a lot of .proto specific looking code for a test.  

We could simplify this further by just creating files named `<name>.txt`  and then generating corresponding .java files with just `<name>`.java.  If you wanted to generate different classes from one txt file then maybe you could just have a test file that lists class names inside.